File: 03.singleton/singleton.py
class Database:
    _instance = None

    def __new__(cls, *args, **kwargs):
        if not cls._instance:
            cls._instance = super().__new__(cls)
        return cls._instance

    def connect(self):
        print("Conexión establecida.")

# Uso
db1 = Database()
db2 = Database()

print(db1 == db2)  # True
db1.connect()

File: 03.singleton/singleton.js
class Database {
    constructor() {
        if (Database.instance) return Database.instance;
        this.connection = "Conexión establecida";
        Database.instance = this;
    }
}
const db1 = new Database();
const db2 = new Database();
console.log(db1 === db2); // true, misma instancia